Sources/NIO/EventLoopFuture.swift

```diff
@@ -891,6 +894,42 @@ extension EventLoopFuture {
         
         return body
     }
+    
+    /// Returns a new `EventLoopFuture` that fires only when all the provided futures complete.
+    /// The new `EventLoopFuture` contains the result of combining the `initialResult` with the
+    /// values of the `[EventLoopFuture<U>]`. This funciton is analogous to the standard library's
+    /// reduce(into:), which does not make copies of the result type for each EventLoopFuture.
+    ///
+    /// The returned `EventLoopFuture` will fail as soon as a failure is encountered in any of the
+    /// `futures`. However, the failure will not occur until all preceding
+    /// `EventLoopFutures` have completed. At the point the failure is encountered, all subsequent
+    /// `EventLoopFuture` objects will no longer be waited for. This function therefore fails fast: once
+    /// a failure is encountered, it will immediately fail the overall EventLoopFuture.
+    ///
+    /// - parameters:
+    ///     - initialResult: An initial result to begin the reduction.
+    ///     - futures: An array of `EventLoopFuture` to wait for.
+    ///     - eventLoop: The `EventLoop` on which the new `EventLoopFuture` callbacks will fire.
+    ///     - combiner: The bifunction used to combine partialResults with new elements.
+    /// - returns: A new `EventLoopFuture` with the combined value.
+    public static func reduce<U>(into initialResult: T, _ futures: [EventLoopFuture<U>], eventLoop: EventLoop, combiner: @escaping (inout T, U) -> Void) -> EventLoopFuture<T> {
```

Comment:
Looking into this a little more, I disagree with the choice of argument labels, especially `eventLoop` and `combiner`/`reducer`. It is fine most of the time, when using trailing closure syntax, but even in the body of `reduce(into:)` it looks unnatural: `combiner(&result, value)`. Function name should be a verb. But that's an implementation details. I would vote for removing the label for the closure parameter completely, and since we're at it, `eventLoop` label seems to be repeating type information, so why not drop as well?

Reply:
That makes sense, how about using names similar to the standard library's reduce (`nextPartialResult`, `updateAccumulatingResult`)?
The current `andAll` implementation in master uses `eventLoop` label, so I was trying to follow the same convention to be consistent. If we want to drop it, I suggest we also remove it from `andAll(_:eventLoop)`. But I am not sure

Reply:
I think let's keep the `eventLoop` label for now. I accept it's a wart, but we'd want to remove it from `andAll` and that's a nice semver major, so let's leave it for now.